contracts/TracerPerpetualSwaps.sol

```diff
@@ -198,55 +197,59 @@ contract TracerPerpetualSwaps is
      */
     function executeTrade(
         Perpetuals.Order memory order1,
-        Perpetuals.Order memory order2,
-        uint256 fillAmount
+        Perpetuals.Order memory order2
     ) internal {
-        // fill amount > 0. Overflow occurs when fillAmount > 2^256 - 1
-        int256 _fillAmount = fillAmount.toInt256();
-        int256 quoteChange =
-            PRBMathUD60x18.mul(fillAmount, order1.price).toInt256();
-
-        // Update account states
+        // Retrieve account state
         Balances.Account storage account1 = balances[order1.maker];
         Balances.Account storage account2 = balances[order2.maker];
 
+        // Construct `Trade` types suitable for use with LibBalances
+        (Balances.Trade memory trade1, Balances.Trade memory trade2) =
+            (
+                Balances.Trade(order1.price, order1.amount, order1.side),
+                Balances.Trade(order2.price, order2.amount, order2.side)
+            );
+
+        bytes32 orderId1 = Perpetuals.orderId(order1);
+        bytes32 orderId2 = Perpetuals.orderId(order2);
+
+        // Calculate new account state
+        (Balances.Position memory newPos1, Balances.Position memory newPos2) =
+            (
+                Balances.applyTrade(
+                    account1.position,
+                    trade1,
+                    Balances.fillAmount(
+                        trade1,
+                        filled[orderId1],
+                        filled[orderId2]
+                    ),
+                    feeRate
+                ),
+                Balances.applyTrade(
+                    account2.position,
+                    trade2,
+                    Balances.fillAmount(
+                        trade2,
+                        filled[orderId1],
+                        filled[orderId2]
+                    ),
+                    feeRate
+                )
+            );
+
```

Comment:
Fill amount should be the `min(order1.amount - filled[orderId1], order2.amount - filled[order2])`. As in, if order1 has 10 units left over, but order2 has 5 units left over, the fillAmount for both sides of the order should be 5 units.

At the moment, it makes the `fillAmount` of one side `min(order1.amount - filled[orderId1], order1.amount - filled[order2])`, and the `fillAmount` of the other side `min(order2.amount - filled[orderId1], order2.amount - filled[order2])`